odxtools/diaglayers/hierarchyelement.py

```diff
@@ -306,34 +305,40 @@ def _compute_available_objects(
             inherited_objects = [
                 x
                 for x in parent_dl._compute_available_objects(get_local_objects, get_not_inherited)
-                if x.short_name not in not_inherited_short_names
+                if x[0].short_name not in not_inherited_short_names
             ]
 
             # update the result set with the objects from the current parent_ref
-            for obj in inherited_objects:
+            for obj, obj_layer, obj_prio in inherited_objects:
 
                 # no object with the given short name currently
                 # exits. add it to the result set and continue
                 if obj.short_name not in result_dict:
-                    result_dict[obj.short_name] = (obj, parent_dl)
+                    result_dict[obj.short_name] = (obj, obj_layer, obj_prio)
                     continue
 
                 # if an object with a given name already exists,
                 # there's no problem if it was inherited from a parent
                 # of different priority than the one currently
                 # considered
-                orig_prio = result_dict[obj.short_name][1].variant_type.inheritance_priority
-                new_prio = parent_dl.variant_type.inheritance_priority
+                orig_prio = result_dict[obj.short_name][2]
+
+                if obj_prio == TInheritancePrio(0):
+                    # special case for inheriting from ECU-SHARED-DATA
+                    new_prio = cast(TInheritancePrio, self.variant_type.inheritance_priority - 0.5)
```

Comment:
why 0.5 ?

Reply:
Objects that are inherited from ECU-SHARED-DATA get a specialization, that is less than the inheriting layer but higher than the next general layer. That's also why i made it a float and not an int.

Reply:
Does this 0.5 accumulate? for example if there are multiple ecu shared data?

Reply:
No accumulation, each object gets its own specialization value.

Reply:
it would be simpler to put priority at int, each layer have its layer multiplied by 10, (0, 10, 20, etc)

and simply increase it here by 5.

reason is that float does not do equality well, see https://en.m.wikipedia.org/wiki/Floating-point_arithmetic#Accuracy_problems